src/saltext/vault/states/vault_pki.py

```diff
@@ -0,0 +1,502 @@
+            file_managed_ret = _file_managed(name, contents=contents, replace=replace, **file_args)
+            _add_sub_state_run(ret, file_managed_ret)
+            if not _check_file_ret(file_managed_ret, ret, file_exists):
+                return ret
+
+    except (CommandExecutionError, SaltInvocationError) as err:
+        ret["result"] = False
+        ret["comment"] = str(err)
+        ret["changes"] = {}
+
+    return ret
+
+
+def role_managed(name, mount="pki", issuer_ref=None, ttl=None, max_ttl=None, **kwargs):
+    """
+    Ensures PKI role is present and configured as required.
+
+    name
+        The name of the role.
+
+    mount
+        The mount path the PKI backend is mounted to. Defaults to ``pki``.
+
+    issuer_ref
+        Issuer reference for the role. Can be name, id or literal ``default``.
+
+    ttl
+        Specifies the Time To Live value to be used for the validity period of the requested certificate,
+        provided as a string duration with time suffix. Hour is the largest suffix.
+        The value specified is strictly used for future validity.
+        If not set, uses the system default value or the value of ``max_ttl``, whichever is shorter.
+
+    max_ttl
+        Specifies the maximum Time To Live provided as a string duration with time suffix.
+        Hour is the largest suffix. If not set, defaults to the system maximum lease TTL.
+
+    kwargs
+        Any other parameter accepted by Vault
+        :obj:`write_role <saltext.vault.modules.vault_pki.write_role>` execution module or Vault update role API method.
+    """
+
+    ret = {
+        "name": name,
+        "result": True,
+        "comment": "The role is present as specified",
+        "changes": {},
+    }
+
+    kwargs = {k: v for k, v in kwargs.items() if not k.startswith("_")}
+
+    def _diff_params(current):
+        nonlocal issuer_ref, ttl, max_ttl, kwargs
+        diff_params = (
+            ("issuer_ref", issuer_ref),
+            ("ttl", timestring_map(ttl, cast=int)),
+            ("max_ttl", timestring_map(max_ttl, cast=int)),
+        )
+        changed = {}
+        for param, arg in diff_params:
+            if arg is None:
+                continue
+            if current[param] != arg:
+                changed.update(
+                    {
+                        param: {
+                            "old": current.get(param),
+                            "new": arg,
+                        }
+                    }
+                )
+        for param, arg in kwargs.items():
+            if param in current and current[param] != arg:
+                changed.update(
+                    {
+                        param: {
+                            "old": current.get(param),
+                            "new": arg,
+                        }
+                    }
+                )
+        return changed
+
+    current = __salt__["vault_pki.read_role"](name, mount=mount)
```

Comment:
This should be moved into the `try..except` block below as well since it can raise exceptions that we don't want to stacktrace from.